docs: add wenqili as a contributor (#91)
ml5js/ml5-website: docs/api-SketchRNN.md

---
id: SketchRNN
title: SketchRNN()
---

SketchRNN is model 

This original paper and implementation of SketchRNN is TensorFlow and [Magenta.js](https://magenta.tensorflow.org/get-started/#magenta-js) was made by [David Ha](https://twitter.com/hardmaru). The ml5.js implementation was ported by [Reiichiro Nakano](https://github.com/reiinakano).

## Example

```javascript
// Create a new Style Transfer Instance

> const style = ml5.styleTransfer('data/myModel/', modelLoaded);

Reviewer: I know this is in progress, just noting that this is the style transfer code.

Reviewer 2: Yep, just there as placeholder

---
id: SketchRNN
title: SketchRNN()
---

> SketchRNN is a recurrent neural network that was trained on millions of doodles collected from the [Quick, Draw! game](https://quickdraw.withgoogle.com/). With SketchRNN you can have a network continue drawing a figure based on initial strokes.

Reviewer: ```suggestion
SketchRNN is a recurrent neural network model trained on millions of doodles collected from the [Quick, Draw! game](https://quickdraw.withgoogle.com/). The SketchRNN model can create new drawings (from a list of categories) based on an initial path.
```

This original paper and implementation of SketchRNN was made in TensorFlow and ported to [Magenta.js](https://magenta.tensorflow.org/get-started/#magenta-js) by [David Ha](https://twitter.com/hardmaru). The ml5.js implementation was ported by [Reiichiro Nakano](https://github.com/reiinakano).

> A list of completes models that you can with SketchRNN can be found [here]()

Reviewer: ```suggestion
The ml5 library includes [a list of supported SketchRNN models](https://github.com/ml5js/ml5-library/blob/master/src/SketchRNN/models.js).
```

(Note that using "here" or "click here" for links isn't good for accessibility!)

## Example

```javascript
// Create a new SketchRNN Instance
const model = ml5.SketchRNN('cat', modelReady);

// When the model is loaded
function modelReady() {
  console.log('SketchRNN Model Loaded!');
}
// Reset the model's current stat

> model.reset();

Reviewer: this should be in `modelReady()`?

// Generate a new stroke
model.generate(gotSketch);

function gotSketch(err, result) {
  // Do something with the result
}
```

> [Here](https://github.com/ml5js/ml5-examples/blob/master/p5js/SketchRNN/sketch.js) is a complete example.

Reviewer: ```suggestion
Here is [a complete example](https://github.com/ml5js/ml5-examples/blob/master/p5js/SketchRNN/sketch.js).
```

## Syntax
  ```javascript
  ml5.SketchRNN(model, ?callback)
  ```

### Parameters

  `model` - The name of the model to use.

  `callback` - Optional. A function to be called once the model is loaded. If no callback is provided, it will return a promise that will be resolved once the model has loaded.

## Properties

  ```javascript
  .ready
  ```
  > Boolean value that specifies if the model has loaded.

## Methods

  ```javascript
  .reset()
  ```

>   > Reset the model's current RNN state

Reviewer: ```suggestion
  > Reset the model's current state
```

  ```javascript
  .generate(?seed, ?options, ?callback)
  ```
  
  > Generates a new sample with the current state.

>   `seed` - Optional. A seed to be passed to the model before generating a new stroke.

Reviewer: Should probably include info about how to format this.

>   `options` - Optional. An object describing the options of the model.

Reviewer: Should probably include more info like with `charRNN`
